api/peripheral_api/routes/RFID.js

```diff
@@ -0,0 +1,131 @@
+const { hashedByte } = require('../util/RFID-helpers');
+const express = require('express');
+const router = express.Router();
+const RFID = require('../models/RFID');
+const {
+  checkIfTokenSent,
+  checkIfTokenValid,
+} = require('../../util/token-functions');
+const { OK, BAD_REQUEST, UNAUTHORIZED, FORBIDDEN, NOT_FOUND } =
+  require('../../util/constants').STATUS_CODES;
+
+const awsIot = require('aws-iot-device-sdk');
+
+const device = awsIot.device({
+  keyPath: '../api/config/AWS-IOT/private.pem.key',
+  certPath: '../api/config/AWS-IOT/cert.pem.crt',
+  caPath: '../api/config/AWS-IOT/AmazonRootCA1.pem',
+  clientId: 'CentauriServer',
+  host: 'ae3c662b19597-ats.iot.us-west-1.amazonaws.com'
+});
+
+let addRfid = false;
+let newName = null;
+
+device
+  .on('connect', function() {
+    /* eslint-disable-next-line */
+    console.log('Connected to AWS IoT!');
+    device.subscribe('MessageForNode');
+  });
+
+device
+  .on('message', async function(topic, payload) {
+    if (addRfid) {
+      const newRFID = new RFID({
+        name: newName,
+        byte: await hashedByte(JSON.parse(payload.toString()).message)
+      });
+      RFID.create(newRFID, (error) => {
+        if (error) {
+          device.publish('MessageForESP32',
+            JSON.stringify({
+              message: BAD_REQUEST
+            }));
+        } else {
+          device.publish('MessageForESP32',
+            JSON.stringify({
+              message: OK
+            }));
+        }
+        newName = null;
+        addRfid = false;
+        clearTimeout();
+      });
+    } else {
+      const {
+        message
+      } = JSON.parse(payload.toString());
+      const hash = await hashedByte(message);
+      RFID.findOne(
```

Comment:
i thought we were iterating over every stored byte and comparing?

Reply:
Completely forgot about that. I will change it to that